From c5cdbbf8660622011822f23c2b79a9e8c94c761e Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Wed, 26 Sep 2012 13:17:50 -0400
Subject: [PATCH] ocf/random: update use of mix_pool_bytes()

The following commit adds an out parameter to mix_pool_bytes(), and
refactors the use of that function, passing NULL in most cases:

upstream stable commit 9233ef75a68d43f82dbd4bf699177c4f9f55a1a3
    random: use lockless techniques in the interrupt path

Follow suit here, where ocf uses mix_pool_bytes() in a new function
random_input_words(), and where random_input_words() is not expecting an
"out".

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 drivers/char/random.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/char/random.c b/drivers/char/random.c
index f1dd4c8..28c2015 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -814,7 +814,7 @@ void add_disk_randomness(struct gendisk *disk)
  */
 void random_input_words(__u32 *buf, size_t wordcount, int ent_count)
 {
-	mix_pool_bytes(&input_pool, buf, wordcount*4);
+	mix_pool_bytes(&input_pool, buf, wordcount*4, NULL);
 
 	credit_entropy_bits(&input_pool, ent_count);
 
-- 
1.7.9.7

